tests/backup/backup_longevity_test.go

```diff
@@ -0,0 +1,428 @@
+	var populateDone bool
+	triggerEventsChan := make(chan *EventRecord, 100)
+	triggerFunctions = map[string]func(*[]*scheduler.Context, *chan *EventRecord){
+		CreatePxBackup:           TriggerCreateBackup,
+		CreatePxBackupAndRestore: TriggerCreateBackupAndRestore,
+		CreateRandomRestore:      TriggerCreateRandomRestore,
+		DeployBackupApps:         TriggerDeployBackupApps,
+	}
+	//Creating a distinct trigger to make sure email triggers at regular intervals
+	emailTriggerFunction = map[string]func(){
+		EmailReporter: TriggerEmailReporter,
+	}
+
+	BeforeEach(func() {
+		if !populateDone {
+			StartPxBackupTorpedoTest("BackupLongevityTest",
+				"Longevity Run For Backup", nil, 0, ATrivedi, Q1FY24)
+			populateIntervals()
+			//  populateDisruptiveTriggers()
+			populateDone = true
+		}
+	})
+
+	It("has to schedule app and introduce test triggers", func() {
+		log.InfoD("schedule apps and start test triggers")
+		watchLog := fmt.Sprintf("Start watch on K8S configMap [%s/%s]",
+			configMapNS, testTriggersConfigMap)
+
+		Step(watchLog, func() {
+			log.InfoD(watchLog)
+			err := watchConfigMap()
+			if err != nil {
+				log.Fatalf(fmt.Sprintf("%v", err))
+			}
+		})
+
+		TriggerDeployBackupApps(&contexts, &triggerEventsChan)
+		TriggerAddBackupCluster(&contexts, &triggerEventsChan)
+		TriggerAddBackupCredAndBucket(&contexts, &triggerEventsChan)
+
+		var wg sync.WaitGroup
+		Step("Register test triggers", func() {
+			for triggerType, triggerFunc := range triggerFunctions {
+				log.InfoD("Registering trigger: [%v]", triggerType)
+				go testTrigger(&wg, &contexts, triggerType, triggerFunc, &triggerLock, &triggerEventsChan)
+				wg.Add(1)
+			}
+		})
+		log.InfoD("Finished registering test triggers")
+		if Inst().MinRunTimeMins != 0 {
+			log.InfoD("Longevity Tests  timeout set to %d  minutes", Inst().MinRunTimeMins)
+		}
+
+		Step("Register email trigger", func() {
+			for triggerType, triggerFunc := range emailTriggerFunction {
+				log.InfoD("Registering email trigger: [%v]", triggerType)
+				go emailEventTrigger(&wg, triggerType, triggerFunc, &emailTriggerLock)
+				wg.Add(1)
+			}
+		})
+		log.InfoD("Finished registering email trigger")
+
+		CollectEventRecords(&triggerEventsChan)
+		wg.Wait()
+		close(triggerEventsChan)
+		Step("teardown all apps", func() {
+			for _, ctx := range contexts {
+				TearDownContext(ctx, nil)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts)
+	})
+})
+
+func testTrigger(wg *sync.WaitGroup,
+	contexts *[]*scheduler.Context,
+	triggerType string,
+	triggerFunc func(*[]*scheduler.Context, *chan *EventRecord),
+	triggerLoc *sync.Mutex,
+	triggerEventsChan *chan *EventRecord) {
+	defer wg.Done()
+
+	minRunTime := Inst().MinRunTimeMins
+	timeout := (minRunTime) * 60
+
+	start := time.Now().Local()
+	lastInvocationTime := start
+
+	for {
+		// if timeout is 0, run indefinitely
+		if timeout != 0 && int(time.Since(start).Seconds()) > timeout {
```

Comment:
This will always be false

Reply:
MinRunTimeMins default value is 0 and in that case this condition will never be true and i.e expected. in case any MinRunTimeMins is given like 60, this condition will be true after 60 mins